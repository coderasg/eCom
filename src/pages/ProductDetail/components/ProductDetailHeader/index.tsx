import React from 'react';
import { Link } from 'react-router-dom';

import OutBoxIcon from './../../../../components/Icons/OutBox/index.svg';
import SearchIcon from './../../../../components/Icons/Search/index.svg';
import BackArrow from './../../../../components/Icons/BackArrow/index.svg';
import { $iconDefaultColor } from './../../../../components/Icons/IconColors';

import styles from './productDetailHeader.module.scss';

type Props = {
  title: string;
};

function ProductDetailHeader(props: Props) {
  const { title } = props;

  return (
    <div
      data-testid="ProductDetailHeader-wrapper"
      className={styles.productDetailHeader}
    >
      <section className={styles.titleSection}>
        <Link to={'/'}>
          <BackArrow width={25} height={25} fill={$iconDefaultColor} />
        </Link>
        <span className={styles.title}>{title}</span>
      </section>
      <section className={styles.searchSection}>
        <SearchIcon width={25} height={25} fill={$iconDefaultColor} />
        <OutBoxIcon width={25} height={25} fill={$iconDefaultColor} />
      </section>
    </div>
  );
}

export default ProductDetailHeader;
