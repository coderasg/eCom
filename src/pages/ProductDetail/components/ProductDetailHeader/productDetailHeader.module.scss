@use './../../../../variables';

.productDetailHeader {
  display: flex;
  align-items: center;

  section {
    flex-grow: 1;
    justify-content: space-around;
  }

  .titleSection {
    display: flex;
    align-items: center;
    font-size: 20px;
    color: variables.$titleColor;
    font-weight: bold;
  }

  .searchSection {
      display: flex;
      align-items: flex-end;
  }
}
