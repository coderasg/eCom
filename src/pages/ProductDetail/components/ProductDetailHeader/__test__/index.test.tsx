import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import ProductDetailHeader from '../index';

describe('(Component) ProductDetailHeader', () => {
  const defaultProps = {
    title: 'test',
  };

  test('should pass snapshot test', () => {
    const wrapper = renderer.create(
      <BrowserRouter>
        <ProductDetailHeader {...defaultProps} />
      </BrowserRouter>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
