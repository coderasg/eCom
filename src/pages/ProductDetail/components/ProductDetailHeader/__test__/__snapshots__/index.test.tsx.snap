// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ProductDetailHeader should pass snapshot test 1`] = `
<div
  data-testid="ProductDetailHeader-wrapper"
>
  <section>
    <a
      href="/"
      onClick={[Function]}
    >
      <SvgIcon
        fill="#b1b0b5"
        height={25}
        width={25}
      />
    </a>
    <span>
      test
    </span>
  </section>
  <section>
    <SvgIcon
      fill="#b1b0b5"
      height={25}
      width={25}
    />
    <SvgIcon
      fill="#b1b0b5"
      height={25}
      width={25}
    />
  </section>
</div>
`;
